lib/schedule.js

```diff
@@ -188,46 +188,59 @@ Job.prototype.schedule = function(spec) {
   var inv;
   var start;
   var end;
+  var timezone = null;
 
   if (typeof spec === 'object' && spec.rule) {
-    start = spec.start || null;
-    end = spec.end || null;
+    start = spec.start || undefined;
+    end = spec.end || undefined;
+    timezone = spec.timezone || null;
     spec = spec.rule;
 
-    if (start != null) {
-      if (!(start instanceof Date)) {
-        start = new Date(start);
+    if (start !== undefined) {
+      if (!(start instanceof CronDate)) {
+        start = new CronDate(start, timezone);
       }
       if (!isValidDate(start) || start.getTime() < Date.now()) {
-        start = null;
+        start = undefined;
       }
+
     }
 
-    if (end != null && !(end instanceof Date) && !isValidDate(end = new Date(end))) {
-      end = null;
+    if (end !== undefined) {
+      if (!(end instanceof CronDate)) {
+        end = new CronDate(end, timezone);
+      }
+      if (!isValidDate(end)) {
+        end = undefined;
+      }
+
     }
+
   }
 
   try {
-    var res = cronParser.parseExpression(spec, { currentDate: start });
+
+    var res = cronParser.parseExpression(spec, { currentDate: start, tz: timezone });
     inv = scheduleNextRecurrence(res, self, start, end);
     if (inv !== null) {
       success = self.trackInvocation(inv);
     }
 
   } catch (err) {
-    var type = typeof spec;
-    if ((type === 'string') || (type === 'number')) {
-      spec = new Date(spec);
+    var typeOfSpec = typeof spec;
+    if (typeOfSpec === 'string' ||
+      typeOfSpec === 'number') {
+      spec = new CronDate(spec, timezone);
     }
 
-    if ((spec instanceof Date) && (isValidDate(spec))) {
-      if (spec.getTime() >= Date.now()) {
+
+    if (spec instanceof CronDate) {
```

Comment:
You have forgotten about the case where `spec` is a `Date`. I would left the code like this:
```js
    if (typeof spec === 'string') {
      spec = new Date(spec);
    }

    if (typeof spec === 'number' ||
        spec instanceof Date) {
      spec = new CronDate(spec, timezone);
    }

    if (spec instanceof CronDate) {
      if(isValidDate(spec) && spec.getTime() >= Date.now()) {
        inv = new Invocation(self, spec);
        scheduleInvocation(inv);
        success = self.trackInvocation(inv);
      }
    } else if (typeof spec ===  'object') {
      if (!(spec instanceof RecurrenceRule)) {
        var r = new RecurrenceRule();
        if ('year' in spec) {
          r.year = spec.year;
        }

       ....
```